src/robusta/core/discovery/discovery.py

```diff
@@ -132,11 +132,20 @@ def discovery_process() -> DiscoveryResults:
         try:
             current_jobs: V1JobList = client.BatchV1Api().list_job_for_all_namespaces()
             for job in current_jobs.items:
-                job_labels: Dict[str, str] = job.spec.selector.match_labels
-                job_pods = [
-                    pod.metadata.name for pod in pod_items
-                    if job_labels.items() <= (pod.metadata.labels or {}).items()
-                ]
+                job_pods = []
+                job_labels = {}
+                if job.spec.selector:
+                    job_labels = job.spec.selector.match_labels
+                elif job.metadata.labels:
+                    job_name = job.metadata.labels.get("job-name", None)
+                    if job_name:
+                        job_labels = {"job-name": job_name}
+
+                if job_labels:  # add job pods only if we found a valid selector
```

Comment:
we should only add pods in the same namespace since we can have the case that `job_labels = {"job-name": job_name}` and you can have the same job running in different namespaces

Reply:
it's always a namespaced query

Reply:
fixing